VitruvioHost/Plugins/Vitruvio/Source/VitruvioEditor/Private/VitruvioCooker.cpp

```diff
@@ -0,0 +1,332 @@
+
+	// Copy Materials
+	const auto PolygonGroups = OriginalMeshDescription->PolygonGroups();
+	for (const auto& PolygonGroupId : PolygonGroups.GetElementIDs())
+	{
+		const FName MaterialName = MeshAttributes.GetPolygonGroupMaterialSlotNames()[PolygonGroupId];
+		int32 Index = Mesh->GetMaterialIndex(MaterialName);
+
+		if (Index != INDEX_NONE)
+		{
+			UMaterialInterface* Material = Mesh->GetMaterial(Index);
+
+			UMaterialInstanceDynamic* DynamicMaterial = Cast<UMaterialInstanceDynamic>(Material);
+			check(DynamicMaterial);
+			UMaterialInstanceConstant* NewMaterial = SaveMaterial(DynamicMaterial, Path, MaterialCache, TextureCache);
+			FName NewSlot = PersistedMesh->AddMaterial(NewMaterial);
+
+			MeshAttributes.GetPolygonGroupMaterialSlotNames()[PolygonGroupId] = NewSlot;
+		}
+	}
+
+	// Build the Static Mesh
+	TArray<const FMeshDescription*> MeshDescriptions;
+	MeshDescriptions.Add(OriginalMeshDescription);
+	PersistedMesh->BuildFromMeshDescriptions(MeshDescriptions);
+	PersistedMesh->PostEditChange();
+	PersistedMesh->MarkPackageDirty();
+
+	// Notify asset registry of new asset
+	FAssetRegistryModule::AssetCreated(PersistedMesh);
+
+	MeshCache.Add(Mesh, PersistedMesh);
+	return PersistedMesh;
+}
+} // namespace
+
+void CookVitruvioActors(TArray<AActor*> Actors)
+{
+	// If there is a previous cooking already ongoing we have to wait until it has completed. This could happen because the last part
+	// of the cooking process is asynchronous.
+	BlockUntilCookCompleted();
+
+	IsCooking = true;
+
+	// Wait until all ongoing generate calls to PRT have finished (might happen if we try to cook before all models of a scene
+	// have been generated).
+	BlockUntilGenerated();
+
+	// Notify on generate completed for VitruvioComponents. Unlike the busy waiting for PRT generate calls which happen async we need to use
+	// a callback here as creating the StaticMeshes from generation also happens on the game thread.
+	ModelsGeneratedHandle = VitruvioModule::Get().OnGenerateCompleted.AddLambda([Actors](int GenerateCalls, int Warnings, int Errors) {
+		if (GenerateCalls == 0)
+		{
+			TSharedRef<SDlgPickPath> PickContentPathDlg = SNew(SDlgPickPath).Title(FText::FromString("Choose location for cooked models."));
+
+			if (PickContentPathDlg->ShowModal() == EAppReturnType::Cancel)
+			{
+				return;
+			}
+			FString CookPath = PickContentPathDlg->GetPath().ToString();
+
+			// Cook actors after all models have been generated and their meshes constructed
+			FScopedSlowTask CookTask(Actors.Num(), FText::FromString("Cooking models..."));
+
+			FMaterialCache MaterialCache;
+			FTextureCache TextureCache;
+			FStaticMeshCache MeshCache;
+
+			for (AActor* Actor : Actors)
+			{
+				CookTask.EnterProgressFrame(1);
+
+				AActor* OldAttachParent = Actor->GetAttachParentActor();
+
+				// Spawn new Actor with persisted geometry
+				AActor* CookedActor = Actor->GetWorld()->SpawnActor<AActor>(Actor->GetActorLocation(), Actor->GetActorRotation());

```

Comment:
Is it possible to name the actor the same as the old one (or similar)?

Reply:
Makes sense, will do.